File: easypermissions/src/main/java/za/co/bwmuller/easypermissions/ActivityListener.java
package za.co.bwmuller.easypermissions;

import android.app.Activity;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

/**
 * Created by Bernhard Müller on 10/14/2016.
 */
public interface ActivityListener {
    void onCreate(@Nullable Bundle savedInstanceState);

    void onPostCreate(@Nullable Bundle savedInstanceState);
    void onResume();
    void onPostResume();
    void onStart();
    void onStop();
    void onPause();
    void onDestroy();

    void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions,
                                           @NonNull int[] grantResults);

    Activity getActivity();
}

File: easypermissions/src/main/java/za/co/bwmuller/easypermissions/ActivityListener.kt
package za.co.bwmuller.easypermissions

import android.os.Bundle
import android.app.Activity

/**
 * Created by Bernhard Müller on 10/14/2016.
 */
interface ActivityListener {

    fun onCreate(savedInstanceState: Bundle?)
    fun onPostCreate(savedInstanceState: Bundle?)
    fun onResume()
    fun onPostResume()
    fun onStart()
    fun onStop()
    fun onPause()
    fun onDestroy()
    fun onRequestPermissionsResult(
        requestCode: Int, permissions: Array<String?>,
        grantResults: IntArray
    )

    val activity: Activity?
}